ast-checks/src/missing-patch-comment.rs

```diff
@@ -0,0 +1,106 @@
+use std::collections::HashMap;
+use std::{env, error::Error, fs};
+
+use rnix::types::{List, TypedNode, Wrapper};
+use rnix::SyntaxNode;
+
+use codespan::{FileId, Files};
+
+use serde::{Deserialize, Serialize};
+
+mod comment_finders;
+mod tree_utils;
+
+use comment_finders::{find_comment_above, find_comment_within};
+use tree_utils::walk_keyvalues_filter_key;
+
+
+#[derive(Serialize, Deserialize, Debug)]
+struct ErrorLocation {
+    column: usize,
+    line: usize,
+    file: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct MissingCommentError {
+    cond: bool,
+    locations: Vec<ErrorLocation>,
+    msg: &'static str,
+    name: &'static str,
+}
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let args: Vec<String> = env::args().skip(1).collect();
+    let mut report = HashMap::new();
+
+    let mut files = Files::new();
+    for filename in args {
+        let file_id = files.add(filename.clone(), fs::read_to_string(&filename)?);
+        report.insert(filename.clone(), execute(&files, file_id)?);
+    }
+
+    println!("{}", serde_json::to_string(&report)?);
+    Ok(())
+}
+
+fn execute(
+    files: &Files<String>,
+    file_id: FileId,
+) -> Result<Vec<MissingCommentError>, Box<dyn Error>> {
```

Comment:
I don't think this makes sense at face value. It returns either a vec of errors or an error? How can this ever succeed?